packages/squiggle-lang/__tests__/SqProject/SqProject_imports_test.ts

```diff
@@ -1,81 +1,221 @@
+        "./common",
+        "./myModule",
+      ]);
+    });
 
-  test("getImports", () => {
-    expect(project.getImports("main")).toEqual({
-      ok: true,
-      value: [
-        { variable: "common", sourceId: "./common" },
-        { variable: "myVariable", sourceId: "./myModule" },
-      ],
+    test("getDependents", () => {
+      expect(project.getDependents("./common")).toEqual(["main"]);
+    });
+    test("getImportIds", () => {
+      const mainImportIds = project.getImportIds("main");
+      if (mainImportIds.ok) {
+        expect(mainImportIds.value).toEqual(["./common", "./myModule"]);
+      } else {
+        throw new Error(mainImportIds.value.toString());
+      }
+    });
+
+    test("getImports", () => {
+      expect(project.getImports("main")).toEqual({
+        ok: true,
+        value: [
+          { type: "named", variable: "common", sourceId: "./common" },
+          { type: "named", variable: "myVariable", sourceId: "./myModule" },
+        ],
+      });
+    });
+
+    test("continues", () => {
+      expect(project.getContinues("main")).toEqual([]);
     });
   });
 
-  test("continues", () => {
-    expect(project.getContinues("main")).toEqual([]);
+  test("Without linker", async () => {
+    const project = SqProject.create();
+    project.setSource(
+      "main",
+      `
+        import './lib' as lib
+        123
+      `
+    );
+
+    await project.run("main");
+
+    expect(project.getResult("main").ok).toEqual(false);
   });
-});
 
-describe("Unknown imports", () => {
-  test("run", async () => {
-    const project = SqProject.create({ resolver: (name) => name });
+  test("Unknown import", async () => {
+    const project = SqProject.create({ linker: buildNaiveLinker() });
     project.setSource(
       "main",
       `
 import './lib' as lib
-123`
+lib.x`
     );
 
     await project.run("main");
-
     expect(project.getResult("main").ok).toEqual(false);
+    expect(project.getResult("main").value.toString()).toEqual(
+      "Failed to load import ./lib"
+    );
   });
 
-  test("runWithImports", () => {
-    const project = SqProject.create({ resolver: (name) => name });
+  test("Known import", async () => {
+    const project = SqProject.create({
+      linker: buildNaiveLinker({
+        "./lib": "export x = 5",
+      }),
+    });
     project.setSource(
       "main",
       `
 import './lib' as lib
 lib.x`
     );
 
-    expect(
-      project.runWithImports("main", async (id) => {
-        throw new Error(`Unknown id ${id}`);
-      })
-    ).rejects.toThrow();
-
-    expect(
-      project.runWithImports("main", async (id) => {
-        if (id === "./lib") {
-          return "x = 5";
-        }
-        throw new Error(`Unknown id ${id}`);
-      })
-    ).resolves.toBe(undefined);
+    await project.run("main");
+
+    expect(project.getResult("main").ok).toEqual(true);
+    expect(project.getResult("main").value.toString()).toEqual("5");
+  });
+
+  describe("Another import test", () => {
```

Comment:
? Can you change the name of this describe block? It contains several tests in it, no clear boundary